### DISCUSSION TITLE
deploying F' on AVR32

### DISCUSSION THREAD

**Discussion by capsulecorplab** [nasa/fprime#813] [Q&A] [answered]
Any chance F' can be deployed on an AVR32 microcontroller?

**Comment by Joshua-Anderson** [comment, marked as answer]
People have successfully run F' on AVR microcontrollers before. You'll have to develop the cmake toolchain to support the target, but F' can theoretically run on just about anything that has a C++ compiler.

**Comment by SterlingPeet** [comment]
Full disclosure: I have a F Prime Support package that supports the ATmega (AVR8) architecture.  It is not yet available for public use, and would likely not be quite what you need for AVR32.

Thanks,

Sterling

Sent from my mobile.

On Jul 8, 2021, at 5:10 PM, Joshua Anderson ***@***.***> wrote:

﻿

People have successfully run F' on AVR microcontrollers before. You'll have to develop the cmake toolchain to support the target, F' can theoretically run on just about anything that has a C++ compiler.

**Reply by capsulecorplab** [reply]
@SterlingPeet Curious, how is progress on the AVR8 support package?

**Reply by SterlingPeet** [reply]
Hi!  The work is ongoing, and I have been unfortunately (in a good way) focused on operational readiness for one mission and operations of another mission, both F Prime related.  GT-1 is on-orbit and uses AVR8 F Prime, but it is based on F Prime 1.3 which did not directly translate to 2.0+ releases.

GT-2 is still in development stages, and the modern AVR8 support is being written for that mission, to support 2.0+ releases.  The support there is mostly complete but should be maturing over the next few months.  We are planning to push the support package developed for GT-2 up to a F Prime community repository when it is ready.
